e2e/src/auction_bid.rs

```diff
@@ -0,0 +1,87 @@
+use ethcontract::web3::types::{H160, H256, U256};
+
+#[derive(Debug)]
+pub struct AuctionBid {
+    order_hash: H256,
+    num_orders: u64,
+    creation_time_stamp: U256,
+    pub solver: H160,
+    objective_value: U256,
+    pub tentative_state: H256,
+    solution_hash: H256,
+}
+
+type PreAuctionBid = (
+    [u8; 32],
+    u64,
+    U256,
+    H160,
+    U256,
+    [u8; 32],
+    [u8; 32],
+    U256,
+    U256,
+);
+
+impl AuctionBid {
```

Comment:
nit: this should probably be `impl From<&'_ PreAuctionBid> for AcutionBid` or even `impl From<PreAuctionBid> for AcutionBid`